rotkehlchen/chain/ethereum/modules/yearn/utils.py

```diff
@@ -0,0 +1,129 @@
+            f'Previous query of yearn vaults returned {num_of_vaults} vaults and last API'
+            f'response had the same amount of vaults. Skipping...',
+        )
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            # since primary key is key/value and those are the same we are updating the last
+            # queried timestamp
+            GlobalDBHandler().set_general_cache_values(
+                write_cursor=write_cursor,
+                key_parts=[GeneralCacheType.YEARN_VAULTS],
+                values=[num_of_vaults[0]],
+            )
+        return
+
+    for vault in data:
+        if 'type' not in vault:
+            log.error(f'Could not identify the yearn vault type for {vault}. Skipping...')
+            continue
+
+        if vault['type'] == 'v1':
+            vault_type = YEARN_VAULTS_V1_PROTOCOL
+        elif vault['type'] == 'v2':
+            vault_type = YEARN_VAULTS_V2_PROTOCOL
+        else:
+            log.error(f'Found yearn token with unknown version {vault}. Skipping...')
+            continue
+
+        try:
+            underlying_token = get_or_create_evm_token(
+                userdb=db,
+                evm_address=string_to_evm_address(vault['token']['address']),
+                chain_id=ChainID.ETHEREUM,
+                form_with_incomplete_data=True,
+                decimals=vault['token']['decimals'],
+                name=vault['token']['name'],
+                symbol=vault['token']['symbol'],
+            )
+
+            vault_token = get_or_create_evm_token(
+                userdb=db,
+                evm_address=string_to_evm_address(vault['address']),
+                chain_id=ChainID.ETHEREUM,
+                protocol=vault_type,
+                form_with_incomplete_data=True,
+                decimals=vault['decimals'],
+                name=vault['name'],
+                symbol=vault['symbol'],
+                underlying_tokens=[UnderlyingToken(
+                    address=underlying_token.evm_address,
+                    token_kind=EvmTokenKind.ERC20,
+                    weight=ONE,
+                )],
+            )
+        except KeyError as e:
+            log.error(
+                f'Failed to store information about tokens for yean {vault_type} vault due to '
+                f'missing key {str(e)}. Vault: {vault}. Skipping...',
+            )
+            continue
+
+        # if it existed but the protocol is not correct edit it. Can happen if it was auto added
+        # before this logic existed or executed.
+        if vault_token.protocol != vault_type:
+            log.debug(f'Editing yearn asset {vault_token}')
+            # we have to use setattr since vault_token is frozen
+            object.__setattr__(vault_token, 'protocol', vault_type)
+            GlobalDBHandler().edit_evm_token(vault_token)
+
+    # Update the cache to mark the vaults as queried
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
```

Comment:
I still don't get how you use this cache or what you write. It's non-obvious from the code, comments and variable names.